typed-protocols/src/Network/TypedProtocol/Driver.hs

```diff
@@ -74,16 +79,30 @@ import Numeric.Natural (Natural)
 -- helpful utility for use in custom drives.
 --
 
-
 --
--- Driver for normal peers
+-- Tracing
 --
 
 -- | Structured 'Tracer' output for 'runPeer' and derivitives.
 --
 data TraceSendRecv ps = TraceSendMsg (AnyMessage ps)
                       | TraceRecvMsg (AnyMessage ps)
 
+-- requires UndecidableInstances extension
+instance Show (AnyMessage ps) => Show (TraceSendRecv ps) where
```

Comment:
I didn't add this precisely because it requires `UndecidableInstances`. I've consulted with colleagues and it's not as bad as I was thinking.

Part of the reason it's not too bad here is that this is the only instance decl in the module, so we should document that here and where we declare the extension, so we don't add more instances later without thinking.